controllers/Datarequest.php

```diff
@@ -15,85 +15,120 @@ public function __construct() {
+        $isDatamanager       = $this->api->call('datarequest_is_role', ["role" => "DM"])->data;
+        $isExecutiveDirector = $this->api->call('datarequest_is_role', ["role" => "ED"])->data;
+
         $viewParams = array(
-            'styleIncludes' => array(
+            'styleIncludes'       => array(
                 'lib/datatables/css/datatables.min.css',
                 'lib/font-awesome/css/font-awesome.css',
                 'css/datarequest/index.css'
             ),
-            'scriptIncludes' => array(
+            'scriptIncludes'      => array(
                 'lib/datatables/js/datatables.min.js',
                 'js/datarequest/index.js',
             ),
-            'items'        => $items,
-            'activeModule' => 'datarequest',
-            'help_contact_name' => $this->config->item('help_contact_name'),
-            'help_contact_email' => $this->config->item('help_contact_email')
+            'items'               => $items,
+            'activeModule'        => 'datarequest',
+            'isProjectManager'    => $isProjectManager,
+            'isExecutiveDirector' => $isExecutiveDirector,
+            'isDatamanager'       => $isDatamanager,
+            'help_contact_name'   => $this->config->item('datarequest_help_contact_name'),
+            'help_contact_email'  => $this->config->item('datarequest_help_contact_email')
         );
 
         loadView('/datarequest/index', $viewParams);
     }
 
-    function datarequestStatus($requestId) {
-	$result = $this->api->call('datarequest_get', ['request_id' => $requestId]);
-	return $result->data->requestStatus;
-    }
-
     public function view($requestId) {
         # Check user group memberships and statuses
-        $isBoardMember  = $this->api->call('datarequest_is_bod_member')->data;
-        $isDatamanager  = $this->api->call('datarequest_is_datamanager')->data;
-        $isDMCMember    = $this->api->call('datarequest_is_dmc_member')->data;
-        $isRequestOwner = $this->api->call('datarequest_is_owner', ['request_id' => $requestId])->data;
-        $isReviewer     = $this->api->call('datarequest_is_reviewer', ['request_id' => $requestId])->data;
+        $isProjectManager    = $this->api->call('datarequest_is_role', ["role" => "PM"])->data;
+        $isExecutiveDirector = $this->api->call('datarequest_is_role', ["role" => "ED"])->data;
+        $isDatamanager       = $this->api->call('datarequest_is_role', ["role" => "DM"])->data;
+        $isDMCMember         = $this->api->call('datarequest_is_role', ["role" => "DMC"])->data;
+        $isRequestOwner      = $this->api->call('datarequest_is_role',
+                                                ["role" => "OWN", "request_id" => $requestId])->data;
+        $isReviewer          = $this->api->call('datarequest_is_role',
+                                                ["role" => "REV", "request_id" => $requestId])->data;
 
```

Comment:
Can we do this more efficient? We now have to do 6 API calls to get the role of the user. Can we get this information when loading the form or combine these and return a single JSON structure with the user group memberships and statuses?

Reply:
True. Will fix.

Reply:
Fixed.